fuse_optimizers/src/optimizer.cpp

```diff
@@ -243,14 +243,18 @@ void Optimizer::loadSensorModels()
 
   for (const auto& config : sensor_model_configs)
   {
+    // Check whether this is an ignition sensor model or not
+    const bool ignition = config.config.hasMember("ignition") ? static_cast<bool>(config.config["ignition"]) : false;
+
     // Create a sensor object using pluginlib. This will throw if the plugin name is not found.
     auto sensor_model = sensor_model_loader_.createUniqueInstance(config.type);
     // Initialize the sensor
     sensor_model->initialize(
       config.name,
       std::bind(&Optimizer::injectCallback, this, config.name, std::placeholders::_1));
     // Store the sensor in a member variable for use later
-    sensor_models_.emplace(config.name, std::move(sensor_model));
+    sensor_models_.emplace(config.name,
+                           SensorModelInfo{ std::move(sensor_model), ignition });  // NOLINT(whitespace/braces)
```

Comment:
I couldn't use `{ std::move(sensor_model), ignition }`, or `{{ std::move(sensor_model), ignition }}`.

So I had to add a constructor to `SensorModelInfo`.

I also had to take the `sensor_model` by copy, but I'm using `std::move` here and in the attribute initialization. Not sure if all the `std::move`s are required, or whether I could take the argument with move semantics using `&&`. :thinking:

Reply:
Best I can tell, passing unique_ptrs by value is the "correct" method. And then enough `std::move` calls are sprinkled around to make it work. I always come back to the [this article](https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/) to figure out how I'm supposed to do things with smart pointers.

Reply:
Good. Then, according to that article, in this case this class constructor implements a "sink", so the implementation looks correct. Thanks.